src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -199,6 +199,50 @@ public override void VisitOperatorDeclaration(OperatorDeclarationSyntax node)
             Visit(node.ExpressionBody);
         }
 
+        public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+        {
+            if (node.MayBeNameofOperator())
+            {
+                var oldEnclosing = _enclosing;
+                WithTypeParametersBinder withTypeParametersBinder = ((_enclosing.Flags & BinderFlags.InContextualAttributeBinder) != 0) && node.IsFeatureEnabled(MessageID.IDS_FeatureExtendedNameofScope)
+                    ? getExtraWithTypeParametersBinder(_enclosing, getAttributeTarget(_enclosing))
+                    : null;
+
+                var argumentExpression = node.ArgumentList.Arguments[0].Expression;
+                var possibleNameofBinder = new NameofBinder(argumentExpression, _enclosing, withTypeParametersBinder);
+                AddToMap(node, possibleNameofBinder);
+
+                _enclosing = possibleNameofBinder;
+                base.VisitInvocationExpression(node);
+                _enclosing = oldEnclosing;
+                return;
+            }
+
+            base.VisitInvocationExpression(node);
+            return;
+
+            static Symbol getAttributeTarget(Binder current)
+            {
+                Debug.Assert((current.Flags & BinderFlags.InContextualAttributeBinder) != 0);
+
+                do
+                {
+                    if (current is ContextualAttributeBinder contextualAttributeBinder)
```

Comment:
A similar traversal is also done in `GetForwardedToAssembly` in `Binder_Symbols.cs`. Should we introduce a virtual method on `Binder` for this? It's fine if the answer is "no", thought I would check either way.